test/e2e/data/infrastructure-docker/bases/cluster-with-kcp.yaml

```diff
@@ -51,6 +51,7 @@ metadata:
   labels:
     kcp-adoption.step2: ""
 spec:
+  nodeDrainTimeout: ${NODE_DRAIN_TIMEOUT}
```

Comment:
If we are applying this to the base `cluster-with-kcp` this means that all the E2E tests based on the default template are going to interrupt drain after 60s. I'm not sure it is ok to change the behavior of all the tests... is there some specific reason why you went down this path instead of refining the usage of a dedicated template?

Reply:
Maybe I didn't get you right. If I understand right, you mentioned above that a small change should be only like a patch to the standard bases?

Reply:
Oh I got your idea. Did you mean to redefine only KCP and machinedeployment to avoid duplicating the whole yaml files?

Reply:
Sorry if I wasn't clear.
My suggestion is to not change base yamls, but to create kustomize patches to change the base for the node drain test following the approach described [here](https://kubectl.docs.kubernetes.io/guides/introduction/kustomize/#2-create-variants-using-overlays). 

An example of a kustomize patch can be found in cluster-api at https://github.com/kubernetes-sigs/cluster-api/blob/ad28a70ab0c3ee804cebc8084fdbb92e10cc24b6/controlplane/kubeadm/config/manager/manager_image_patch.yaml (this patche changes the  image field, but you can use the same approach for adding nodeDrainTimeout)

In other words, IMO.
- it is fine to create a new template for node-drain
- this template should be created by composing existing bases and then applying overlay patches to add the nodeDrainTimeout field where required (instead of duplicating the whole bases)

Reply:
Got it. Thanks